_posts/2018-03-08-roadmap.md

```diff
@@ -0,0 +1,221 @@
+
+Part of the goal with the Rust 2018 release is to provide high quality
+documentation for the full set of new and improved features and the idioms they
+give rise to. [The Rust Programming Language book][trpl] has been completely
+re-written over the last 18 months, and will be updated throughout the year as
+features reach the stable compiler. [Rust By Example] will likewise undergo a
+revamp this year. And there are numerous third party books, like [Programming
+Rust], reaching print as well.
+
+[trpl]: https://doc.rust-lang.org/nightly/book/second-edition/
+[Programming Rust]: https://www.google.com/search?q=programming+rust&ie=utf-8&oe=utf-8&client=firefox-b-1-ab
+[Rust By Example]: https://rustbyexample.com/
+
+**Language improvements**
+
+The most prominent language work in the pipeline stems from [2017’s ergonomics
+initiative]. Almost all of the accepted RFCs from the initiative are available
+on nightly today, and will be polished and stabilized over the next several
+months. Among these productivity improvements are a few “headliners” that will
+form the backbone of the release:
+
+[2017’s ergonomics initiative]: https://blog.rust-lang.org/2017/03/02/lang-ergonomics.html
+
+-   **Ownership system improvements**, including making borrowing more flexible
+    via “non-lexical lifetimes”, improved pattern matching integration, and
+    more.
+-   **Trait system improvements**, including the long-awaited impl Trait syntax
+    for dealing with types abstractly.
+-   **Module system improvements**, focused on increasingly clarity and reducing
+    complexity.
+-   **Generators/async/await:** work is rapidly progressing on first-class async
+    programming support.
+
+In addition, we anticipate a few more major features to stabilize prior to the
+epoch release, including **SIMD**, **custom allocators**, and **macros 2.0**.
+
+### Compiler improvements
+
+As of [Rust 1.24](https://blog.rust-lang.org/2018/02/15/Rust-1.24.html),
+incremental recompilation is available and enabled by default on the stable
+compiler. This feature already makes rebuilds significantly faster than fresh
+builds, but over the course of the year we expect continued improvements for
+_both_ fresh and re-builds. Compiler performance should not be an obstacle to
+productivity in Rust 2018.
+
+
+
+### Tooling improvements
+
+Rust 2018 will see high quality 1.0 releases of the RLS (underlying much of our
+IDE integration story) and rustfmt (a standard formatting tool for Rust
+code). We will continue to improve Cargo by stabilizing custom registries,
+public dependencies, and a revised profile system. We’re also expecting further
+work on [Cargo build system integration], [Xargo integration], and [custom test
+frameworks], though it’s unclear as yet which of these will be complete prior to
+the epoch release.
+
+[Cargo build system integration]: https://github.com/rust-lang/rfcs/pull/2136
+[Xargo integration]: https://github.com/japaric/xargo
```

Comment:
i think this should link to https://github.com/rust-lang/cargo/issues/4959, as it's way less confusing